src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -93,4 +93,69 @@ public static void FileSystemWatcher_ChangeWatchedFolder()
             Utility.ExpectEvent(eventOccured, "changed");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Changed_NestedDirectories()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
+            {
+                // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                System.Threading.Thread.Sleep(1000);
```

Comment:
I get nervous any time I see explicit sleeps being put in code like this, for two reasons:

1. Under certain conditions the pause time could end up being too low, resulting in sporadic failures.
2. The longer the forced pause time, the longer the test takes.  We have a goal (we're not there yet) of each suite of inner loop tests taking no more than 5 seconds, and just from the sleeps added in this PR, the running time for this suite will increase almost 20 seconds.

Instead, how about just blocking here until the event is fired for the creation of firstDir?  Create the child directory, wait for the event, create the next child, wait for the event, etc.  I'm ok with having long timeouts on those waits because they're not forcing a normal execution to slowdown, but rather just providing some wiggle room in case something's running a little slower here and there.  This is why in some of our tests you see things like ```Assert.True(someCondition.Wait(LongTimeout));``` (where LongTimeout is 30 seconds or something)... usually the condition being waited for will happen immediately, and it should always happen, but we give it 30 seconds wiggle room just in case the lab is running very slowly or something, enough time that if the assert ever does fail, something is definitely wrong.